src/python/pants_release/generate_github_workflows.py

```diff
@@ -884,6 +887,53 @@ def macos14_arm64_test_jobs() -> Jobs:
     return jobs
 
 
+def windows11_x86_64_test_jobs() -> Jobs:
+    helper = Helper(Platform.WINDOWS11_X86_64)
+    jobs = {
+        helper.job_name("build"): {
+            "name": "Test in-progress Windows support",
+            "runs-on": helper.runs_on(),
+            "timeout-minutes": 60,
+            "if": IS_PANTS_OWNER,
+            "steps": [
+                *checkout(),
+                {
+                    "name": "Install MSYS2",
+                    "uses": "msys2/setup-msys2@v2",
+                    "with": {
+                        "msystem": "UCRT64",
+                        "install": "base-devel mingw-w64-x86_64-toolchain mingw-w64-ucrt-x86_64-nasm mingw-w64-x86_64-cmake mingw-w64-ucrt-x86_64-protobuf",
+                    },
+                },
+                {
+                    "name": "Set Up Rust Toolchain",
+                    "uses": "actions-rust-lang/setup-rust-toolchain@v1",
```

Comment:
For my education, why can't rustup be used on Windows here?

Also, use `action` function for common action references.